ui/components/src/TimeChart/TimeChart.tsx

```diff
@@ -270,10 +267,28 @@ export const TimeChart = forwardRef<ChartInstance, TimeChartProps>(function Time
     __experimentalEChartsOptionsOverride,
     noDataVariant,
     timeZone,
-    tooltipPinnedCoords,
     isStackedBar,
+    enablePinning,
+    pinnedCrosshair,
   ]);
 
+  // Update adjacent charts so tooltip is unpinned when current chart is clicked.
+  useEffect(() => {
+    // Only allow pinning one tooltip at a time, subsequent tooltip click unpins previous.
+    // Multiple tooltips can only be pinned if Ctrl or Cmd key is pressed while clicking.
+    if (tooltipPinnedCoords !== null && lastTooltipPinnedCoords !== null) {
+      if (!isEqual(lastTooltipPinnedCoords, tooltipPinnedCoords)) {
+        setTooltipPinnedCoords(null);
+      }
+      const isCrosshairPinned = checkCrosshairPinnedStatus(seriesMapping);
```

Comment:
Should this be `updatedSeriesMapping`?

Reply:
This useEffect is for resetting pinned tooltip and crosshairs in synched charts so calling setPinnedCrosshair and setTooltipPinnedCoords should be sufficient to trigger the useMemo that ultimately updates seriesMapping inside of it.

I am looking into adjusting the reset behavior though (see [comment](https://github.com/perses/perses/pull/1317/#issuecomment-1650545480))